### PR TITLE
 feat(Paddle-frontend):  Adds paddle.fft.ihfft2

### LINKED ISSUE #23759: ihfft2

### PR DESCRIPTION
**Pull request by firi193**
The ihfft2 function transforms a real array frequency representation of a signal into a hermitian symmetric complex array of the signal in the time domain in two dimensions. This can be used to understand and better analyze the frequency of a signal, as the real number array frequency format is better suited for this

Close #23759

## Checklist

-  [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [ ] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py [hunk 1]
```diff
@@ -180,6 +181,40 @@ def test_paddle_ifftshift(
...
+    frontend,
+    backend_fw,
+    test_flags,
+    fn_tree,
+):
+    input_dtypes, x, axis = dtype_x_axis
+    shape = ivy.array(x).shape
+    x = ivy.reshape(x, shape)  # reshape into a 2 dimensional array.
+
+    for norm in ["backward", "forward", "ortho"]:
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py:203]: The `norm` value should be generated using the `hypothesis` strategy.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py [hunk 2]
```diff
@@ -180,6 +181,40 @@ def test_paddle_ifftshift(
...
+        ivy.astype(
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py:204]: This is redundant as in the function test body we only need to call `helpers.test_frontend_function(*args, **kwargs)`

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @firi193 

Thanks for contributing to Ivy😊
Can you remove the changes that you added to the files in `.idea` directory
Also check the comments below👇

Happy coding 🤞

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py (hunk 2 above)
Comment (firi193) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py:204]: Hello Hirwa, thank you for leaving a review. I will leave out the for loop line then, "for norm in ["backward", "forward", "ortho"]:".  So, is the ihfft2 function in the fft.py alright?

**Comment by firi193** [comment]
> Hey @firi193
> 
> Thanks for contributing to Ivy😊 Can you remove the changes that you added to the files in `.idea` directory Also check the comments below👇
> 
> Happy coding 🤞



> Hey @firi193
> 
> Thanks for contributing to Ivy😊 Can you remove the changes that you added to the files in `.idea` directory Also check the comments below👇
> 
> Happy coding 🤞

@hirwa-nshuti, I have only made the .idea files that were different in my branch to be the same as the ones in the ivy main branch. Should I still undo the changes?

**Comment by hirwa-nshuti** [comment]
> > Hey @firi193
> > Thanks for contributing to Ivy😊 Can you remove the changes that you added to the files in `.idea` directory Also check the comments below👇
> > Happy coding 🤞
> 
> > Hey @firi193
> > Thanks for contributing to Ivy😊 Can you remove the changes that you added to the files in `.idea` directory Also check the comments below👇
> > Happy coding 🤞
> 
> @hirwa-nshuti, I have only made the .idea files that were different in my branch to be the same as the ones in the ivy main branch. Should I still undo the changes?

Yeah only the changes to your PR should be the subtask that you're working on
For more, you can follow this guide https://unify.ai/docs/ivy/overview/contributing/the_basics.html#managing-your-fork

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py [hunk 3]
```diff
@@ -258,6 +260,40 @@ def test_paddle_ifftshift(
...
+    frontend,
+    backend_fw,
+    test_flags,
+    fn_tree,
+):
+    input_dtypes, x, axis = dtype_x_axis
+    shape = ivy.array(x).shape
+    x = ivy.reshape(x, shape)  # reshape into a 2 dimensional array.
+
+    ivy.astype(
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py:283]: We don't need to call `ivy` here only the `helpers.test_frontend_function` will run the tests

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py [hunk 4]
```diff
@@ -258,6 +260,40 @@ def test_paddle_ifftshift(
...
+    dtype_x_axis,
+    norm,
+    frontend,
+    backend_fw,
+    test_flags,
+    fn_tree,
+):
+    input_dtypes, x, axis = dtype_x_axis
+    shape = ivy.array(x).shape
+    x = ivy.reshape(x, shape)  # reshape into a 2 dimensional array.
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py:281]: This should be handled when generating examples in the `dtype_x_axis` argument above

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @firi193 

Check the comment below 😊
Thanks

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py (hunk 4 above)
Comment (firi193) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py]: Hey @hirwa-nshuti, I have left out the `for norm in ["backward", "forward", "ortho"]` line, and also lines 280, and 281, where I have previously called `ivy` to reshape the input. All 5 tests have passed when I run them locally as well. 

Thank you for your time!

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # global
 from hypothesis import given, strategies as st
 
+
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py:4]: This line should be removed as it is redudant
```suggestion
```

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py [hunk 6]
```diff
@@ -258,6 +259,50 @@ def test_paddle_ifftshift(
     )
 
 
+@handle_frontend_test(
+    fn_tree="paddle.fft.ihfft2",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=["float64"],
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py:265]: Can we sample all supported dtypes as suggested by the function wrapper on the implementation side

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @firi193 

Just a few changes are needed here, after that we will be able to merge the PR🤞

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py (hunk 6 above)
Comment (firi193) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py:265]: When I used all the supported dtypes in the test function, I was getting an Assertion error. Something like "returned dtype is complex64 but ground truth returned dtype is complex128 ". Then when I cast the return type to complex128, the assertion error switches into "returned dtype is complex128 but ground truth returned dtype is complex64"(just interchanging the places of complex64 and complex128). 

Is it better if I change the supported dtypes on the implementation side to only "float64", or include all the supported dtypes in the test function wrapper(@handle_frontend_test) even if the assertion error will show up?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py (hunk 5 above)
Comment (firi193) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py:4]: sure I will

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py (hunk 6 above)
Comment (hirwa-nshuti) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py]: This should be handled by promoting due to correct precision `complex` dtype. You can maintain the conversion of result to be `complex64` if input is `int32`, `float32` else if the input is `float64` and `int64` we can cast the result to `complex128`

Comment (firi193) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_fft.py]: Hey @hirwa-nshuti, I have included all the supported types in the test function and handled the conversion of the correct precision complex dtype for each corresponding input dtype as you suggested. All 5 tests are passing as well.

**Review by hirwa-nshuti — APPROVED** [review]
Looks great to me😊
Thank you for contributing🙏

**Comment by firi193** [comment]
> Looks great to me😊 Thank you for contributing🙏

Thank you so much for your cooperation!

**Pull request merged**